tests/baselines/reference/excessPropertyCheckWithUnions.errors.txt

```diff
@@ -108,9 +106,8 @@ excessPropertyCheckWithUnions.ts(112,63): error TS2322: Type 'string' is not ass
             a: "a",
             b: "b", // excess -- kind: "A"
             ~
-!!! error TS2322: Type '{ kind: "A"; n: { a: string; b: string; }; }' is not assignable to type 'AB'.
-!!! error TS2322:   Types of property 'n' are incompatible.
-!!! error TS2322:     Object literal may only specify known properties, and 'b' does not exist in type 'AN'.
+!!! error TS2353: Object literal may only specify known properties, and 'b' does not exist in type 'AN'.

```

Comment:
This feels like a straight improvement - this was supposed to look like this since https://github.com/microsoft/TypeScript/pull/55152 (cc @RyanCavanaugh - maybe it would be worth rechecking if the approach used by that PR doesn't have any other holes)